automation/deploy-bin.ts

```diff
@@ -15,70 +15,135 @@
+};
 
-				const outputStream = fs.createWriteStream(outputFile);
+const finalReleaseAssets: { [platform: string]: string[] } = {
+	win32: [standaloneZips['win32'], renamedOclifInstallers['win32']],
+	darwin: [standaloneZips['darwin'], renamedOclifInstallers['darwin']],
+	linux: [standaloneZips['linux']],
+};
+
+/**
+ * Create the zip file for the standalone 'pkg' bundle previously created
+ * by the buildPkg() function in 'build-bin.ts'.
+ */
+export async function zipStandaloneInstaller() {
+	const outputFile = standaloneZips[process.platform];
+	if (!outputFile) {
+		throw new Error(
+			`Standalone installer unavailable for platform "${process.platform}"`,
+		);
+	}
+	await mkdirpAsync(path.dirname(outputFile));
+	await new Bluebird((resolve, reject) => {
+		console.log(`Zipping build to "${outputFile}"...`);
+
+		const archive = archiver('zip', {
+			zlib: { level: 7 },
+		});
+		archive.directory(path.join(ROOT, 'build-bin'), 'balena-cli');
 
-				outputStream.on('close', resolve);
-				outputStream.on('error', reject);
+		const outputStream = fs.createWriteStream(outputFile);
 
-				archive.on('error', reject);
-				archive.on('warning', console.warn);
+		outputStream.on('close', resolve);
+		outputStream.on('error', reject);
 
-				archive.pipe(outputStream);
-				archive.finalize();
-			}),
-	)
-	.then(() => {
-		console.log('Build zipped');
-		console.log('Publishing build...');
+		archive.on('error', reject);
+		archive.on('warning', console.warn);
 
-		return publishReleaseAsync({
+		archive.pipe(outputStream);
+		archive.finalize();
+	});
+	console.log('Build zipped');
+}
+
+/**
+ * Create or update a release in GitHub's releases page, uploading the
+ * installer files (standalone zip + native oclif installers).
+ */
+export async function createGitHubRelease() {
+	console.log(`Publishing release ${version} to GitHub`);
+	const ghRelease = await Bluebird.fromCallback(
+		publishRelease.bind(null, {
 			token: GITHUB_TOKEN || '',
 			owner: 'balena-io',
 			repo: 'balena-cli',
 			tag: version,
 			name: `balena-CLI ${version}`,
 			reuseRelease: true,
-			assets: [outputFile],
-		});
-	})
-	.then(release => {
-		console.log(`Release ${version} successful: ${release.html_url}`);
-	})
-	.catch(err => {
+			assets: finalReleaseAssets[process.platform],
+		}),
+	);
+	console.log(`Release ${version} successful: ${ghRelease.html_url}`);
+}
+
+/**
+ * Top-level function to create a CLI release in GitHub's releases page:
+ * call zipStandaloneInstaller(), rename the files as we'd like them to
+ * display on the releases page, and call createGitHubRelease() to upload
+ * the files.
+ */
+export async function release() {
+	console.log(`Creating release assets for CLI ${version}`);
+	try {
+		await zipStandaloneInstaller();
+	} catch (error) {
+		console.log(`Error creating standalone installer zip file: ${error}`);
+		process.exit(1);
+	}
+	if (process.platform === 'win32' || process.platform === 'darwin') {
```

Comment:
Why is this synchronous?

Reply:
Done (I didn't even need to import `mz` for promises, as `deploy-bin` itself was already importing `fs-extra`!)